src/helpers/paywall-variables-helpers.ts

```diff
@@ -0,0 +1,367 @@
+  total_price_and_per_month_full: string | number | undefined;
+  sub_price_per_month: string | number | undefined;
+  sub_price_per_week: string | number | undefined;
+  sub_duration: string | number | undefined;
+  sub_duration_in_months: string | number | undefined;
+  sub_period: string | number | undefined;
+  sub_period_length: string | number | undefined;
+  sub_period_abbreviated: string | number | undefined;
+  sub_offer_duration: string | number | undefined;
+  sub_offer_duration_2: undefined;
+  sub_offer_price: string | number | undefined;
+  sub_offer_price_2: undefined;
+  sub_relative_discount: string | number | undefined;
+  [key: string]: string | number | undefined;
+};
+
+enum AbbreviatedPeriod {
+  month = "mo",
+  year = "yr",
+  week = "wk",
+  day = "d",
+}
+
+function getPricePerPeriod(
+  formattedPrice: string,
+  product: SubscriptionOption,
+  full: boolean = false,
+) {
+  return `${formattedPrice}/${product.base.period?.number}${full ? (product.base.period as Period).unit : AbbreviatedPeriod[(product.base.period as Period).unit as keyof typeof AbbreviatedPeriod]}`;
+}
+
+function getDurationInMonths(period: Period) {
+  if (period.unit === "week" || period.unit === "day") {
+    return `${period.number} ${period.unit}`;
+  }
+  if (period.unit === "month") {
+    return `${period.number > 1 ? period.number + "s" : period.number} ${period.unit}`;
+  }
+  if (period.unit === "year") {
+    return `${period.number * 12} ${period.unit}`;
+  }
+}
+
+function getPricePerWeek({
+  price,
+  period,
+  selectedLocale,
+}: {
+  price: Price;
+  period: Period | null;
+  full?: boolean;
+  selectedLocale: string;
+}) {
+  const fallback = formatPrice(
+    price.amountMicros,
+    price.currency,
+    selectedLocale,
+  );
+
+  if (!period) return fallback;
+
+  if (period.unit === "year") {
+    return formatPrice(
+      price.amountMicros / 12 / 4,
+      price.currency,
+      selectedLocale,
+    );
+  }
+  if (period.unit === "month") {
+    return formatPrice(
+      price.amountMicros / period.number,
+      price.currency,
+      selectedLocale,
+    );
+  }
+  if (period.unit === "week" && period.number > 1) {
+    return formatPrice(
+      price.amountMicros / period.number,
+      price.currency,
+      selectedLocale,
+    );
+  }
+  if (period.unit === "day") {
+    return formatPrice(
+      (price.amountMicros * 7) / period.number,
+      price.currency,
+      selectedLocale,
+    );
+  }
+
+  return fallback;
+}
+
+function getPricePerMonth({
+  price,
+  period,
+  selectedLocale,
+}: {
+  price: Price;
+  period: Period | null;
+  full?: boolean;
+  selectedLocale: string;
+}) {
+  const fallback = formatPrice(
+    price.amountMicros,
+    price.currency,
+    selectedLocale,
+  );
+  if (!period) return fallback;
+
+  if (period.unit === "year") {
+    return formatPrice(price.amountMicros / 12, price.currency, selectedLocale);
+  }
+  if (period.unit === "month" && period.number > 1) {
```

Comment:
Maybe you can early return here
```suggestion
  if(period.number == 0)
  {
     return fallback;
  }

  if (period.unit === "month" && period.number > 1) {
```

And avoid the following checks